src/main/java/org/kiwiproject/registry/consul/server/ConsulHelpers.java

```diff
@@ -0,0 +1,80 @@
+package org.kiwiproject.registry.consul.server;
+
+import static org.kiwiproject.base.KiwiPreconditions.checkArgumentNotNull;
+import static org.kiwiproject.registry.util.Ports.findFirstPortPreferSecure;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.experimental.UtilityClass;
+import org.kiwiproject.base.DefaultEnvironment;
+import org.kiwiproject.base.KiwiEnvironment;
+import org.kiwiproject.registry.config.ServiceInfo;
+import org.kiwiproject.registry.model.Port.PortType;
+import org.kiwiproject.registry.model.ServiceInstance;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Set of utilities that assist in populating a service registration.
+ */
+@UtilityClass
+public class ConsulHelpers {
+
+    /**
+     * Creates a map with default metadata items that the {@link org.kiwiproject.registry.consul.client.ConsulRegistryClient}
+     * will be looking for when it converts a registered service back into a {@link org.kiwiproject.registry.model.ServiceInstance}.
+     * <p>
+     * This map is intended to be a baseline of metadata to register.  This map is modifiable so it can easily be
+     * changed or added to later in the registration process.
+     *
+     * @param serviceInfo the information about the service to pull data from.
+     * @return a map that is prepopulated with defaults.
+     */
+    public static Map<String, String> buildDefaultMetaDataMap(ServiceInfo serviceInfo) {
```

Comment:
Since the word _metadata_ is one single word, I suggest we change all the places using camel case `metaData` to just `metadata`. And we attempt to be consistent from here out everywhere else too (other libs, services, etc.)

I also think we sometimes use "buildXxx" and other times "newXxx". I generally prefer "newXxx" for factory methods, b/c I think it's more clear it is creating a new instance. Thoughts?

So this would be renamed as either `buildDefaultMetadataMap` or, applying my other suggestion, `newDefaultMetadataMap`...and then all the other places we have `metaData` would also be renamed.

Reply:
Note this will also be consistent with `ServiceInstance#metadata`